runelite-client/src/main/java/net/runelite/client/plugins/statusorbs/StatusOrbsPlugin.java

```diff
@@ -93,67 +106,224 @@
 		AGILITY_CAPE, AGILITY_CAPET, MAX_CAPE
 	);
 
+
+	private static final BufferedImage HEART_NORMAL;
+	private static final BufferedImage HEART_DISEASE;
+	private static final BufferedImage HEART_POISON;
+	private static final BufferedImage HEART_VENOM;
+	static
+	{
+		HEART_NORMAL = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-NORMAL.png"), 26, 26);
+		HEART_DISEASE = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-DISEASE.png"), 26, 26);
+		HEART_POISON = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-POISON.png"), 26, 26);
+		HEART_VENOM = ImageUtil.resizeCanvas(ImageUtil.getResourceStreamFromClass(StatusOrbsPlugin.class, "1067-VENOM.png"), 26, 26);
+	}
+
+	private static final int SPEC_REGEN_TICKS = 50;
+	private static final int NORMAL_HP_REGEN_TICKS = 100;
+
 	@Inject
 	private Client client;
 
 	@Inject
-	private OverlayManager overlayManager;
+	private ClientThread clientThread;
 
 	@Inject
-	private RunEnergyOverlay energyOverlay;
+	private StatusOrbsConfig config;;
 
 	@Inject
-	private RunEnergyConfig energyConfig;
+	private StatusOrbsOverlay overlay;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Getter
+	private double hitpointsPercentage;
+
+	@Getter
+	private double specialPercentage;
+
+	// HeartDisplay
+	private BufferedImage currentHeart;
+
+	// RegenMeter
+	private int ticksSinceSpecRegen;
+	private int ticksSinceHPRegen;
+	private boolean wasRapidHeal;
 
+	// Run Energy
 	private boolean localPlayerRunningToDestination;
 	private WorldPoint prevLocalPlayerLocation;
 
 	@Provides
-	RunEnergyConfig getConfig(ConfigManager configManager)
+	StatusOrbsConfig provideConfig(ConfigManager configManager)
 	{
-		return configManager.getConfig(RunEnergyConfig.class);
+		return configManager.getConfig(StatusOrbsConfig.class);
 	}
 
 	@Override
 	protected void startUp() throws Exception
 	{
-		overlayManager.add(energyOverlay);
+		overlayManager.add(overlay);
```

Comment:
You should add a migrate method that sets the new keys to their old values (if it exists) and then deletes the old key.